File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares(private val n: Int) {

    fun sumOfSquares(): Int = n * (n+1) * (2*n + 1) / 6

    fun squareOfSum(): Int {
        val intermediate = n * (n + 1) / 2
        return intermediate * intermediate
    }

    fun difference(): Int = squareOfSum() - sumOfSquares()
}

File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(n: isize) isize {
    const sumN = @divTrunc(n * (n + 1), 2);
    return sumN * sumN;
}

pub fn sumOfSquares(n: isize) isize {
    return @divTrunc(n * (n+1) * (2*n + 1), 6);
}

pub fn differenceOfSquares(number: isize) isize {
    return squareOfSum(number) - sumOfSquares(number);
}
